frontend/src/metabase/core/hooks/use-form-submit/use-form-submit.ts

```diff
@@ -1,36 +1,38 @@
-import { useCallback } from "react";
+import { useCallback, useState } from "react";
 import type { FormikHelpers } from "formik";
+import { FormState } from "metabase/core/context/FormContext";
 import { FormError } from "./types";
 
 export interface UseFormSubmitProps<T> {
   onSubmit: (values: T, helpers: FormikHelpers<T>) => void;
 }
 
 export interface UseFormSubmitResult<T> {
+  state: FormState;
   handleSubmit: (values: T, helpers: FormikHelpers<T>) => void;
 }
 
 const useFormSubmit = <T>({
   onSubmit,
 }: UseFormSubmitProps<T>): UseFormSubmitResult<T> => {
+  const [state, setState] = useState<FormState>({ status: "idle" });
+
   const handleSubmit = useCallback(
     async (data: T, helpers: FormikHelpers<T>) => {
       try {
-        helpers.setStatus({ status: "pending" });
+        setState({ status: "pending" });
         await onSubmit(data, helpers);
-        helpers.setStatus({ status: "fulfilled" });
+        setState({ status: "fulfilled" });
```

Comment:
My first thought was "why is `setState` called without `this`?" 😄 

![giphy](https://user-images.githubusercontent.com/17258145/201104367-954bed09-44fc-4942-9148-faeb38dbddb4.gif)